src/composables/notifications.ts

```diff
@@ -0,0 +1,227 @@
+  const fetchedNotificationsOffset = ref(0);
+
+  const notificationSettingsCheckedTypes = computed<string[]>(
+    () => notificationSettings.value
+      .filter((setting: INotificationSetting) => setting.checked)
+      .map((s) => s.type),
+  );
+
+  const notificationsAll = computed<INotification[]>(
+    () => [...notificationsSuperhero.value, ...notificationsWallet.value],
+  );
+
+  const notificationsFiltered = computed<INotification[]>(
+    () => notificationsAll.value
+      .filter(
+        ({ status, type }) => (
+          status === NOTIFICATION_STATUS_READ
+          || notificationSettingsCheckedTypes.value.includes(type)
+        ),
+      )
+      .sort((a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime()),
+  );
+
+  const notificationsToShow = computed<INotification[]>(
+    () => notificationsFiltered.value
+      .slice(0, fetchedNotificationsOffset.value * fetchedNotificationsLimit),
+  );
+
+  async function fetchAllNotifications() {
+    const fetchUrl = `${activeNetwork.value.backendUrl}/notification/user/${account.value.address}`;
+    const responseChallenge = await fetchJson(fetchUrl);
+    const signedChallenge = Buffer.from(
+      await sdk.value.signMessage(responseChallenge.challenge),
+    ).toString('hex');
+
+    const respondChallenge = {
+      challenge: responseChallenge.challenge,
+      signature: signedChallenge,
+    };
+    const url = new URL(fetchUrl);
+    Object.entries(respondChallenge).forEach(([key, value]) => url.searchParams.append(key, value));
+    return fetchJson(url.toString());
+  }
+
+  async function modifyNotifications(
+    ids: number[],
+    status: NotificationStatus,
+  ) {
+    if (!ids.length) return;
+    const backendMethod = async (body: any) => postJson(`${activeNetwork.value.backendUrl}/notification`, { body });
+
+    const responseChallenge = await backendMethod({ ids, status, author: account.value.address });
+    const signedChallenge = Buffer.from(
+      await sdk.value.signMessage(responseChallenge.challenge),
+    ).toString('hex');
+    const respondChallenge = {
+      challenge: responseChallenge.challenge,
+      signature: signedChallenge,
+    };
+
+    await backendMethod(respondChallenge);
+  }
+
+  function normalizeNotification(chainNames: IChainNames) {
+    return ({
+      entityId,
+      sourceId,
+      entityType,
+      sender,
+      ...other
+    }: INotification) => ({
+      sourceId,
+      entityId,
+      entityType,
+      ...other,
+      sender,
+      chainName: sender && chainNames?.[sender],
+      path: entityType === NOTIFICATION_ENTITY_TYPE_TIP
+        ? `${AGGREGATOR_URL}tip/${entityId}`
+        : `${AGGREGATOR_URL}tip/${sourceId}/comment/${entityId}`,
+    });
+  }
+
+  async function getSuperheroNotifications() {
+    const chainNames = useState('chainNames');
+
+    notificationsSuperhero.value = (
+      await fetchAllNotifications()
+    ).map(normalizeNotification(chainNames.value));
```

Comment:
`normalizeNotification` is not used anywhere so I suggest to use it's code directly here.